src/state/store.ts

```diff
@@ -0,0 +1,136 @@
+import { IFieldContext, IField, FieldRegistry } from '../data';
+
+export interface ISubscriber {
+    (): void
+}
+
+export interface FieldEditingHandler {
+    (field: IField): void;
+}
+
+// createStore is a helper function for initializing
+// the store.
+export function createStore(props: IStoreProps) {
+    return new Store(props);
+}
+
+export interface IStore {
```

Comment:
You could do that. From my opinion, it would be easier to develop/maintain a general store object that is responsible for handling all shared state between the react-dynamic-formbuilder components and distributing it to them much like Redux